src/prefect/cli/run.py

```diff
@@ -623,6 +617,12 @@ def run(
 
     # Backend flow run -----------------------------------------------------------------
 
+    if schedule and (flow_or_group_id or name):
+        raise ClickException(
+            "Received a local only flag for a non-local flow; "
+            "cannot specify both `--schedule` and `--id`"
+        )
+
```

Comment:
This error message is a bit confusing if they've just specified name. I'd just do
```suggestion
    if schedule:
        raise ClickException("`--schedule` can only be specified for local flow runs")
```

There's a `return` above this so we're definitely not in a local-run anymore and you don't need to check the other variables either.